client/app/controllers/collection/add.js

```diff
@@ -83,43 +11,125 @@ export default Ember.Controller.extend({
     }),
 
     widgets: [],
+    actions: [],
+
+    // Fire enabled actions.
+    updateState: function(actions) {
```

Comment:
core logic in here. iterates over each action and runs the ones that are enabled.